user/user.go

```diff
@@ -75,7 +91,17 @@ func UpdateUser(ctx echo.Context) error {
 	}
 	user.ID = uid
 
-	if u := model.FindUser(user); u.ID != "" {
+	// 既存のuser情報取得
+	thisUser := model.FindUserByUid(user)
```

Comment:
このuserにはRequestに乗ってる更新したい値が入ってると思うから既存のユーザーはとれないんじゃないかね？
```suggestion
	currentUser:= model.FindUserByUid(&model.User{ID: uid})
```
みたいになるイメージやな～

で、user not foundを返す時のロジックは currentUserがnullだったらとかかな？

Reply:
と思ったら、
https://github.com/hiroki-kondo-git/dayMemo_api_go/blob/0bd9dd3d172a1c184e9e868628a7aaac5e57a64d/model/user.go#L40
の内部でuidを参照してるのね！

リーダブルコードっていう本読むと書いてあったりするんやけど、メソッドはその役割がわかるような命名と、正しい引数と返り値を持つのが良いんよね。今回だと命名はいいんだけど、引数が抽象的で使う側が間違える可能性があるから、引数はuidだけとかのほうが良いかもしれない！

Reply:
ふむふむす